Include/pyexpat.h

```diff
@@ -3,7 +3,7 @@
 
 /* note: you must import expat.h before importing this module! */
 
-#define PyExpat_CAPI_MAGIC  "pyexpat.expat_CAPI 1.1"
+#define PyExpat_CAPI_MAGIC  "pyexpat.expat_CAPI 1.2"
```

Comment:
Should it be changed? It can break third-party users of the pyexpat capsule.

Reply:
@serhiy-storchaka I believe it's a protection mechanism against loading non-matching binaries, catching and preventing an accidental mix-up of files not belonging to the same build of Python. I got the idea of bumping this from commit cb5778f00ce48631c7140f33ba242496aaf7102b and if that was right in bumping at addition of a new member, then I should bump version on addition of a new member, too, or so I thought (and still do). What do you think?

Reply:
Previous additions of a new member did not bump it. If new members always added at the end, it should be binary compatible.

I have doubts about cb5778f00ce48631c7140f33ba242496aaf7102b too, but there might be additional reasons in this particular case.

Reply:
@serhiy-storchaka I checked again how that value is used in practice, it's here:

https://github.com/python/cpython/blob/0c80da4c14d904a367968955544dd6ae58c8101c/Modules/_elementtree.c#L4363-L4372

Based on that code my understanding is that loading bigger structs from the future is considered okay unless `PyExpat_CAPI_MAGIC` differs, which implies that (1) magic needs to stay the same to support loading the future from the past and (2) that `PyExpat_CAPI_MAGIC` should only ever change for changes that do not increase size (but keep it constant).

To summarize, that supports your argument.
I checked branches 3.8, 3.9, 3.10, 3.11, 3.12 now and they all have the same comparison code there so reverting that magic string bump will support seamless backporting. I will revert that version bump with the next push in a minute :+1:

Reply:
Pushed.